cmake/Modules/FindComputeCpp.cmake

```diff
@@ -358,31 +454,40 @@ endfunction()
 #  Adds a SYCL compilation custom command associated with an existing
 #  target and sets a dependancy on that new command.
 #
-#  targetName : Name of the target to add a SYCL to.
-#  binaryDir : Intermediate directory to output the integration header.
-#  sourceFiles : Source files to be compiled for SYCL.
+#  TARGET : Name of the target to add SYCL to.
+#  BINARY_DIR : Intermediate directory to output the integration header.
+#  SOURCES : Source files to be compiled for SYCL.
 #
-function(add_sycl_to_target targetName binaryDir sourceFiles)
-
-  set(sourceFiles ${sourceFiles} ${ARGN})
-  set(fileCounter 0)
-  target_include_directories(
-    ${targetName} SYSTEM
-    PRIVATE ${OpenCL_INCLUDE_DIR}
-    PRIVATE ${COMPUTECPP_INCLUDE_DIRECTORY}
+function(add_sycl_to_target)
+  set(options)
+  set(one_value_args
+    TARGET
+    BINARY_DIR
+  )
+  set(multi_value_args
+    SOURCES
+  )
+  cmake_parse_arguments(SDK_ADD_SYCL
+    "${options}"
+    "${one_value_args}"
+    "${multi_value_args}"
+    ${ARGN}
   )
+  set(fileCounter 0)
   # Add custom target to run compute++ and generate the integration header
-  foreach(sourceFile ${sourceFiles})
+  foreach(sourceFile ${SDK_ADD_SYCL_SOURCES})
     if(NOT IS_ABSOLUTE ${sourceFile})
       set(sourceFile "${CMAKE_CURRENT_SOURCE_DIR}/${sourceFile}")
     endif()
-    __build_spir(${targetName} ${sourceFile} ${binaryDir} ${fileCounter})
+    __build_ir(
+      TARGET     ${SDK_ADD_SYCL_TARGET}
+      SOURCE     ${sourceFile}
+      BINARY_DIR ${SDK_ADD_SYCL_BINARY_DIR}
+      COUNTER    ${fileCounter}
+    )
     MATH(EXPR fileCounter "${fileCounter} + 1")
   endforeach()
-
-  # Link with the ComputeCpp runtime library
-  target_link_libraries(${targetName} PUBLIC $<$<OR:$<CONFIG:Debug>,$<CONFIG:RelWithDebInfo>>:${COMPUTECPP_RUNTIME_LIBRARY_DEBUG}>
-                                             $<$<NOT:$<OR:$<CONFIG:Debug>,$<CONFIG:RelWithDebInfo>>>:${COMPUTECPP_RUNTIME_LIBRARY}>
-                                             ${OpenCL_LIBRARIES})
-
+  target_link_libraries(${SDK_ADD_SYCL_TARGET}
+    PUBLIC ComputeCpp::ComputeCpp
```

Comment:
This may be a feature for a future PR, but (optionally?) appending ComputeCpp to the target's `BUILD_RPATH` might be a useful addition so users won't have to set `LD_LIBRARY_PATH` to test their applications.

Reply:
This happens for executables built on the machine, but if you install them they'll be stripped, as far as I can tell.